storybook/stories/API/ResponsiveContainer.mdx

```diff
@@ -0,0 +1,66 @@
+import { ArgTypes } from '@storybook/blocks';
+import * as ResponsiveContainerStories from './ResponsiveContainer.stories';
+
+# ResponsiveContainer
+
+This is a component that can be used to render charts that will be responsive to its parent.
+When you render a chart you have to specify its **width** and **height** for the chart to be
+created correctly and show up when rendered. If you want to render a chart that one or both
+of the dimensions is not a hard-coded number, instead it can fill the container, or be half
+of its parent, then ResponsiveContainer is the right tool for you.
+
+ResponsiveContainer required to have either **width** or **height** or both be a percentage, like
+`75%`, if you're trying to specify both width and height to be just a number, then its highly
+likely that you don't need ResponsiveContainer and you can render the chart as is.
+
+```jsx
+<ResponsiveContainer width="100%" height={400}>
+  <LineChart {...args}>
+    <Line dataKey="uv" />
+  </LineChart>
+</ResponsiveContainer>
+```
+
+Its worth noting that by default, ResponsiveContainer does not render when you're rendering on
+the server (SSR) or if JavaScript is disabled. This is because it depends on browser APIs
+like [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) to
+first calculate what the percentage equals to in terms of its parent element. For example
+if you specify `width="80%"`, ResponsiveContainer should first calculate the width of the
+parent, say its `720px`, now it has to calculate 80 percent of that 720 which will be **576px**
+and only then, renders the child Chart and passes the calculated width to it. So when you're
+rendering a Chart inside ResponsiveContainer, you can omit its width and height and specify
+those on the ResponsiveContainer.
+
+If you're using ResponsiveContainer and you're rendering on the server (SSR) as most JavaScript
+frameworks do, then initially on the Browser, the chart will not be shown until after hydration.
```

Comment:
Nit: "as most JavaScript frameworks do" can probably be removed. I would still say most JavaScript frameworks/libraries render on the client as opposed to the server. Can just call out SSR but not be specific